apisix/plugins/log-rotate.lua

```diff
@@ -277,6 +277,7 @@ local function rotate()
         max_size = attr.max_size or max_size
         enable_compression = attr.enable_compression or enable_compression
     end
+    local timeout = interval * 100 -- Timeout for compression is interval in milliseconds
```

Comment:
We'd better add a new configuration item to control the timeout parameter.

<img width="802" alt="image" src="https://user-images.githubusercontent.com/6814606/210908411-53c1ad81-f9ba-4b11-90bb-a0eea139c6bf.png">

Reply:
I don't agree that timeout should be a user defined parameter. Because, setting a timeout greater than interval may cause undefined behaviour (eg. second rotation starts but first compression is still running), and setting the value too low also causes the mentioned undesired behaviour (so doesn't fix the issue in the first place). So, this value should be automatically set based on interval and shoud not be user defined imho.

Reply:
@alptugay, I agree with you, however setting a fixed timeout is inappropriate. What if the compression starts just before the next rotation?

The correct way would be to provide a timeout such that timeout occurs before the next log-rotation. And if the compression still gets a timeout, we can reschedule the compression later using ngx_timer. WDYT?

Reply:
> What if the compression starts just before the next rotation?
Rotation is done in intervals. Execution step is like below in simple terms:
1) Rename current log file to logfile.date
2) Send signal to process so new file opens 
3) compress logfile.date with a timeout that is not bigger than the interval,
So I don't think step 3 will start just before next rotation. 

But your second argument is valid, compression can still get a timeout if it can't be finished in the specified interval. We can schedule it with ngx_timer but this means those timers will add up. Assume the interval is 5 minutes and compression takes 10 minutes, after 1 hour we will have done 12 intervals jobs but only 6 compression will be completed and 6 will be waiting. So this will add up and probably make the system unresponsive

Reply:
Makes sense. LGTM.

Reply:
I suggest adding a configuration to control the timeout parameter, the default timeout is 10s, it's enough for most people, if you use interval as timeout, it could be 1min or 1 hour, maybe someone don't need such long time.